### PR TITLE
execute CoreBuild to generate xaml files

### LINKED ISSUE #2779: MSBuildWorkspace and WPF Projects
I'm trying to compile a WPF project using `MSBuildWorkspace`. Should this work out of the box, or do I need to do anything special to set it up?

When I open a .sln containing a WPF project and try to compile I receive two errors:

```
C:\Users\Josh\Documents\GitHub\Orange\WpfApplication1\WpfApplication1\MainWindow.xaml.cs(25,13): error CS0103: The name 'InitializeComponent' does not exist in the current context
```
and

```
error CS5001: Program does not contain a static 'Main' method suitable for an entry point
```

To reproduce, simply create a new WPF project. (Mine was WPFApplication1.sln). Then just load into a workspace and try to compile the project.

```CSharp
var msws = MSBuildWorkspace.Create(new Dictionary<string, string> { { "CheckForSystemRuntimeDependency", "true" } });
var soln = msws.OpenSolutionAsync(@"C:\Users\Josh\Documents\GitHub\Orange\WpfApplication1\WpfApplication1.sln").Result;
var comp = soln.Projects.First().GetCompilationAsync().Result;
var errs = comp.GetDiagnostics().Where(n => n.Severity == DiagnosticSeverity.Error).ToList();
//Two errors :(
```

### PR DESCRIPTION
**Pull request by mattwar**
Fixes #2779 

This enables the MSBuild workspace to run the xaml tasks that generate files.  It doesn't solve the redirect problem that must aslo be done by each exe using msbuild vis msbuild-workspace.

@Pilchie @jasonmalinowski  please review

### CODE REVIEW

File: src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs [hunk 1]
```diff
@@ -54,8 +54,18 @@ public string GetPropertyValue(string name)
...
+            // The executed project will hold the final model of the project after execution via msbuild.
+            var executedProject = _loadedProject.CreateProjectInstance();
+
+            // if the project doesn't have a Compile target, then its not really a vb or c# project.
+            if (!executedProject.Targets.ContainsKey("Compile"))
+            {
+                return executedProject;
+            }
+
+            // prepare for building  -- use CoreBuild to get xaml generated files
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs]: As far as I know the normal IntelliSense build in VS just runs Compile...why do we need something different here?

Comment (mattwar) [inline src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs]: The "MarkupCompilePass1" target that compiles the xaml into the generated files is not part of Compile.. It's comes under "PrepareResources" that is only part of CoreBuild.

**Comment by mattwar** [comment]
@dotnet-bot test this please

File: src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs (hunk 1 above)
Comment (Pilchie) [inline src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs]: We should talk to some msbuild people (@aarnott maybe?) to understand what the right thing to do here is.

**Comment by mattwar** [comment]
moved to stabilization PR #2982

**Pull request closed without merging**